pkg/db/metal/switch.go

```diff
@@ -141,3 +142,238 @@ func FromBGPState(state BGPState) (apiv2.BGPState, error) {
 	}
 	return apiv2State, nil
 }
+
+func (s *Switch) ConnectMachine(machine *Machine) (int, error) {
+	_, connectionExists := s.MachineConnections[machine.ID]
+	physicalConnections := s.getPhysicalMachineConnections(machine)
+
+	if len(physicalConnections) < 1 {
+		if connectionExists {
+			return 0, fmt.Errorf("machine connection between machine %s and switch %s exists in the database but not physically; if you are attempting migrate the machine from one rack to another delete it first", machine.ID, s.ID)
+		}
+		return 0, nil
+	}
+
+	delete(s.MachineConnections, machine.ID)
+	s.MachineConnections[machine.ID] = append(s.MachineConnections[machine.ID], physicalConnections...)
+	return len(physicalConnections), nil
+}
+
+// TranslateNicMap creates a NicMap where the keys are translated to the naming convention of the target OS
+//
+// example mapping from cumulus to sonic for one single port:
+//
+//	map[string]Nic {
+//		"swp1s1": Nic{
+//			Name: "Ethernet1",
+//			MacAddress: ""
+//		}
+//	}
+func (s *Switch) TranslateNicMap(targetOS SwitchOSVendor) (NicMap, error) {
+	nicMap := s.Nics.MapByName()
+	translatedNicMap := make(NicMap)
+
+	if s.OS.Vendor == targetOS {
+		return nicMap, nil
+	}
+
+	ports := make([]string, 0)
+	for name := range nicMap {
+		ports = append(ports, name)
+	}
```

Comment:
_Completely optional_

```suggestion
	ports := slices.Collect(maps.Keys(nicMap))
```

Or

```suggestion
    ports := lo.Keys(nicMap)
```